source/agora/test/QuorumPreimage.d

```diff
@@ -0,0 +1,436 @@
+
+    enum quorums_1 = [
+        // 0
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        // 1
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        // 2
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        // 3
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        // 4
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        // 5
+        QuorumConfig(5, [
+            WK.Keys.D.address,
+            WK.Keys.F.address,
+            WK.Keys.B.address,
+            WK.Keys.A.address,
+            WK.Keys.C.address,
+            WK.Keys.E.address]),
+
+        QuorumConfig.init,
+        QuorumConfig.init,
+    ];
+
+    version (none)
+    foreach (idx, node; nodes.enumerate)
+        writefln("Node %s: %s\n", idx, node.getQuorumConfig);
+
+    nodes.enumerate.each!((idx, node) =>
+        retryFor(node.getQuorumConfig() == quorums_1[idx], 5.seconds,
+            format("Node %s has quorum config %s. Expected: %s",
+                idx, node.getQuorumConfig(), quorums_1[idx])));
+
+    Transaction makePayTx (in Transaction prev, PublicKey[] addresses, uint index = 0)
+    {
+        auto input = Input(hashFull(prev), index);
+
+        Transaction tx =
+        {
+            TxType.Payment,
+            [input],
+            addresses.map!(addr => Output(Amount.MinFreezeAmount, addr)).array
+        };
+
+        auto signature = WK.Keys.Genesis.secret.sign(hashFull(tx)[]);
+        tx.inputs[0].signature = signature;
+        return tx;
+    }
+
+    Transaction makeFreezeTransaction (in Transaction prev, PublicKey address)
+    {
+        auto input = Input(hashFull(prev), 0);
+
+        Transaction tx =
+        {
+            TxType.Freeze,
+            [input],
+            [Output(Amount.MinFreezeAmount, address)]
+        };
+
+        auto signature = WK.Keys.Genesis.secret.sign(hashFull(tx)[]);
+        tx.inputs[0].signature = signature;
+        return tx;
+    }
+
+    auto spendable = network.blocks[$ - 1].txs
+        .filter!(tx => tx.type == TxType.Payment)
+        .map!(tx => iota(tx.outputs.length)
+        .map!(idx => TxBuilder(tx, cast(uint)idx)))
```

Comment:
That indentation is triggering me

Reply:
> suggestion pls

Reply:
The `map` is on `iota`, not on the result of the previous `map`, so +4 indent
```suggestion
            .map!(idx => TxBuilder(tx, cast(uint)idx)))
```

Reply:
Oh right. That was a mistake, it's easy to miss.